include/SFML/Window/WindowBase.inl

```diff
@@ -53,20 +54,49 @@ struct DelayOverloadResolution
     {
     }
 };
+
+template <typename Handler>
+struct Caller : Handler
+{
+    using Handler::operator();
+};
+
+template <typename Handler>
+struct Caller<Handler&>
+{
+    template <typename Argument, std::enable_if_t<std::is_invocable_v<Handler, Argument>, int> = 0>
+    decltype(auto) operator()(Argument&& argument)
+    {
+        return handler(std::forward<Argument>(argument));
+    }
+    Handler& handler;
+};
+
+template <typename Return, typename Argument>
+struct Caller<Return (*)(Argument)>
+{
+    Return operator()(Argument&& argument)
+    {
+        return function(std::forward<Argument>(argument));
+    }
+    Return (*function)(Argument);
+};
```

Comment:
I want to make sure I understand what's going on here so let me know if the following summary is correct:

The `Caller<Handler&>` class template specialization exists to create a closure for a reference to a lambda. This enables the use of move-only lambdas because it means we can simply hold a reference to said lambdas instead of actually moving them. 

The `struct Caller<Return (*)(Argument)>` class template specialization is the same idea but for function pointers.

And then there's the non-specialized class template which catches all other callable which do not need special handling.

Reply:
You understand correctly.
You are right, it deserves a few comments. I will add some.

Reply:
Done